libraries/botbuilder/src/cloudAdapter.ts

```diff
@@ -145,6 +144,35 @@ export class CloudAdapter extends CloudAdapterBase implements BotFrameworkHttpAd
         }
     }
 
+    /**
+     * Exposes the same functionality as [BotFrameworkAdapter.processActivityDirect](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder/src/botFrameworkAdapter.ts).
+     *
+     * @param authorization Authorization in format: "Bearer [longString]".
+     * @param activity The activity to process.
+     * @param logic The logic function to apply.
+     * @returns a promise representing the asynchronous operation.
+     */
+    async processActivityDirect(
+        authorization: String,
+        activity: Activity,
+        logic: (context: TurnContext) => Promise<void>
+    ): Promise<void> {
+        const validatedActivity = validateAndFixActivity(ActivityT.parse(activity));
+
+        if (!validatedActivity.type) {
+            console.warn('BadRequest: Missing activity or activity type.');
+            throw new Error('Missing activity or activity type');
+        }
+
+        const authHeader = z.string().parse(authorization ?? '');
+
+        try {
+            await this.processActivity(authHeader, validatedActivity, logic);
+        } catch (err) {
+            throw new Error(err);
```

Comment:
This doesn't have much sense. We are catching the error to throw it again. Also, be careful with stack exposure (codeQL). I'm not sure if it applies here though.